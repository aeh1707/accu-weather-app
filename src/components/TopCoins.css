.top-coins {
  background-color: black;
  justify-content: center;
  display: flex;
  flex-wrap: wrap;
}

a {
  text-decoration: none;
}

a:nth-child(odd) {
  background-color: #486aad;
}

a:nth-child(even) {
  background-color: #6b7a98;
}

.coin-brief {
  /* border: 1px solid #eeedeb; */
  color: #eeedeb;
  max-width: 170px;
  padding: 20px;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.coin-image-container {
  min-height: 175px;
}

.coin-image {
  width: 140px;
  margin: 0 auto;
}

.coin-symbol {
  font-size: 1.7rem;
  font-weight: 700;
}
