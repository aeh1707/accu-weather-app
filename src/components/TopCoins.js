import './TopCoins.css';
import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchCoins, selectFilteredCoins } from '../redux/cryptoReducer';
import Search from './Search';

const TopCoins = () => {
  const dispatch = useDispatch();
  const topCoins = useSelector(selectFilteredCoins);
  useEffect(() => {
    dispatch(fetchCoins());
  }, [dispatch]);
  return (
    <>
      <Search />
      <main className="top-coins">
        {topCoins.map((coin) => (
          <Link
            to={`/coininfo/${coin.id}`}
            key={coin.id}
          >
            <article className="coin-brief">
              <div className="coin-image-container">
                <img className="coin-image" src={coin.image} alt={coin.name} />
              </div>
              <div className="coin-brief-info">
                <div className="coin-symbol">{coin.symbol.toUpperCase()}</div>
                <div className="coin-price">
                  $
                  {coin.current_price}
                </div>
              </div>
            </article>
          </Link>
        ))}
      </main>
    </>

  );
};

export default TopCoins;
