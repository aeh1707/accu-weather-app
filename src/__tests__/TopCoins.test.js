/*eslint-disable*/
import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import store from '../redux/configureStore';
import TopCoins from '../components/TopCoins';

test('TopCoins component testing', () => {
  const TREE = TestRenderer.create(
    <Provider store={store}>
      <TopCoins />
    </Provider>,
  );
  expect(TREE).toMatchSnapshot();
});
