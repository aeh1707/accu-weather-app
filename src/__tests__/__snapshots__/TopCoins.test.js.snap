// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopCoins component testing 1`] = `
Array [
  <div
    className="search"
  >
    <div
      className="search-container"
    >
      <input
        id="search"
        onChange={[Function]}
        placeholder="ex: bitcoin"
        type="text"
        value=""
      />
    </div>
  </div>,
  <main
    className="top-coins"
  />,
]
`;
